__init__.py

```diff
@@ -10,158 +10,116 @@
-            self.settings["use_selene"] = True
-        if "api_key" not in self.settings:
-            self.settings["api_key"] = "Y7R353-9HQAAL8KKA"
-        self.translator = google_translator()
-        self.tx_cache = {}  # avoid translating twice
-        self.answer_cache = {}  # avoid hitting wolfram twice for same question
-        self.selene_api = WAApi()
-
-    def initialize(self):
-        self.blacklist_default_skill()
-
-    def blacklist_default_skill(self):
-        # load the current list of already blacklisted skills
-        blacklist = self.config_core["skills"]["blacklisted_skills"]
-
-        # check the folder name (skill_id) of the skill you want to replace
-        skill_id = "fallback-wolfram-alpha.mycroftai"
-
-        # add the skill to the blacklist
-        if skill_id not in blacklist:
-            self.log.debug("Blacklisting official mycroft skill")
-            blacklist.append(skill_id)
-
-            # load the user config file (~/.mycroft/mycroft.conf)
-            conf = LocalConf(USER_CONFIG)
-            if "skills" not in conf:
-                conf["skills"] = {}
-
-            # update the blacklist field
-            conf["skills"]["blacklisted_skills"] = blacklist
-
-            # save the user config file
-            conf.store()
-
-        # tell the intent service to unload the skill in case it was loaded already
-        # this should avoid the need to restart
-        self.bus.emit(Message("detach_skill", {"skill_id": skill_id}))
-
-    def translate(self, utterance, lang_tgt=None, lang_src="en"):
-        lang_tgt = lang_tgt or self.lang
-
-        # if langs are the same do nothing
-        if not lang_tgt.startswith(lang_src):
-            if lang_tgt not in self.tx_cache:
-                self.tx_cache[lang_tgt] = {}
-            # if translated before, dont translate again
-            if utterance in self.tx_cache[lang_tgt]:
-                # get previous translated value
-                translated_utt = self.tx_cache[lang_tgt][utterance]
-            else:
-                # translate this utterance
-                translated_utt = self.translator.translate(utterance,
-                                                           lang_tgt=lang_tgt,
-                                                           lang_src=lang_src).strip()
-                # save the translation if we need it again
-                self.tx_cache[lang_tgt][utterance] = translated_utt
-            self.log.debug("translated {src} -- {tgt}".format(src=utterance,
-                                                              tgt=translated_utt))
-        else:
-            translated_utt = utterance.strip()
-        return translated_utt
-
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.idx = 0
+        self.last_query = None
+        self.results = []
+        self.wolfie = WolframAlphaSolver({
+            "units": self.config_core['system_unit'],
```

Comment:
We'll need to consider how to handle this per-user/query

Reply:
that will be a future PR, but essentially we have to keep self.results per user. Also needed for proper hivemind support

one thing to note is that we cant create Solver objects all the time, it introduces a lot of latency, so changing units per user should just modify the solver property